src/chunk_insert_state.c

```diff
@@ -69,6 +72,25 @@ create_chunk_result_relation_info(ChunkDispatch *dispatch, Relation rel, Index r
 }
 
 /*
+ * Infer a chunk's set of arbiter indexes. This is a subset of the chunk's
+ * indexes that match the ON CONFLICT statement.
+ */
+static List *
+chunk_infer_arbiter_index(int rindex, List *rtable, Query *parse)
```

Comment:
It is copied, although not a deep copy. But the pointers I change is in the struct copy and I don't think `infer_arbiter_indexes` modifies the query.

Besides, the query itself is not changed, just some of the state/metadata inside the internal struct.